src/codec/CMakeLists.txt

```diff
@@ -6,6 +6,7 @@ cmake_policy(SET CMP0015 NEW)
 
 add_library(${CODEC_TARGET} SHARED
 	CodecMain.cpp
+	CodecMain.h
```

Comment:
now we add headers for the codec target, but not for the serlio target itself, that's inconsistent.
while your at it, how about introducing a query for the current generator and only add the headers for VS generators (via an additional target_sources call)? i would like to keep headers out of the targets for non-VS generators like ninja and make...